### PR TITLE
account script setup when invoking hide()

### PR DESCRIPTION
**Pull request by metalsadman**
https://github.com/quasarframework/quasar/issues/10821

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `dev` branch (or `v[X]` branch)
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Review by yusufkandemir — APPROVED** [review]
This solution is hacky, the real problem behind is that components using `script setup` are [closed by default](https://v3.vuejs.org/api/sfc-script-setup.html#defineexpose), and this doesn't work for this case:
https://github.com/quasarframework/quasar/blob/f138b43de8a74bbd514bd2fecbc28ec89eb5f2c2/ui/src/composables/use-dialog-plugin-component.js#L26-L27
But we can't also simply expect users to call `defineExpose`. Firstly, we don't return `show` and `hide` from the composable. Secondly, even if we return them it makes the usage more complicated than it should be, IMO. I am playing around with the Expose API these days, and will probably have a PR after this gets merged.

In short, since this is basically the same solution that is applied to the `show` method, which is working for a long time, it's fine, we can directly merge it.
